src/app/static/src/app/store/projects/actions.ts

```diff
@@ -92,7 +92,10 @@ export const actions: ActionTree<ProjectsState, RootState> & ProjectsActions = {
         await api<RootMutation, ProjectsMutations>(context)
             .withSuccess(RootMutation.SetProject, true)
             .withError(ProjectsMutations.ProjectError)
-            .postAndReturn<string>("/project/", qs.stringify(payload));
+            .postAndReturn<string>("/project/", qs.stringify(payload))
+            .then(() => {
+                dispatch("getProjects")
+            })
```

Comment:
I also discovered that validation wasn't working when a project is newly created and users try to upload again from zip. Dispatching getProjects updates the previous projects hence validation works.